File: day-12/pi.py
from decimal import Decimal
import math


def pi_func(i, depth):
    if depth == 1:
        return 0
    else:
        return Decimal('1') + i / (Decimal('2.0') * i + Decimal('1')) * pi_func(i + Decimal('1'), depth - 1)  # noqa


pi = 2 * pi_func(Decimal('1'), 100)
print(math.pi)
print(pi)

File: day-12/pi.clj
(defn pi-func
  ([]
   (pi-func 1000 1M))
  ([depth]
   (pi-func depth 1M))
  ([depth i]
   (if (= 0 depth)
     0M
     (+ 1M
        (* (with-precision 100 (/ i (+ 1M (* 2.0M i))))
           (pi-func (dec depth) (inc i)))))))

(def pi (* 2M (pi-func)))
(println Math.PI)
(println pi)
